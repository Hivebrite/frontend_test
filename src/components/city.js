import React from 'react';
import styled from 'styled-components';

const Button = styled.button`
    width: 100%;
    height: 60px !important;
    border: 1px solid lightgrey;
    border-radius: 5px;
    text-align: center;
    margin-bottom: 10px;
    cursor: pointer;
    font-family: 'Helvetica';
    font-weight: bold;
    font-size: 14px;

    &:hover {
        background-color: lightblue;
        color: white;
    }
`;

const City = (props) => {

    return (
        <Button onClick={props.onChoice}>
            {props.name}
        </Button>
    )
};

export default City;

