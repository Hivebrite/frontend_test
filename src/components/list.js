import React, { useEffect, useState } from 'react';
import styled from 'styled-components'

import data from '../data/cities.json';
import City from './city';

const ListWrapper = styled.div`
    width: 20%;
    height: 90%;
    padding: 30px;
    border: 1px solid lightgrey;
    background-color: green;
    border-radius: 3px;
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
`

const SearchInput = styled.input`
    width: 100%;
    padding: 5px;
    border: 1px solid lightgrey;
    border-radius: 3px;
    margin-bottom: 10px;
`
const List = () => {
    const [cities, setCities] = useState([]);
    const [filteredList, setFilteredList] = useState([]);
    const [search, setSearch] = useState('');

    const onFiltering = (e) => {
        setSearch(e.target.value);
    }

    useEffect(() => {
        setCities(data);
    }, []);

    useEffect(() => {
        if (cities && cities.length > 0) {
            console.log('CITIES:', cities.length);
            const temp = cities.slice(0, 10);
            console.log('list', temp);
            setFilteredList(temp);
        }
    }, [cities]);

    return (
        <>
            <ListWrapper>
                <SearchInput type="text" value={search} onChange={onFiltering} placeholder="Filter the list"/>
                {filteredList && filteredList.length > 0 && filteredList.map((element) => {
                    return <City name={element.city}/>
                })}
            </ListWrapper>
        </>
    )

};

export default List;

// const getData = () => {
//     fetch('./test.json'
//         , {
//             headers: {
//                 'Content-Type': 'application/json',
//                 'Accept': 'application/json'
//             }
//         }
//     )
//         .then((response) => {
//             console.log('response', response)
//             return response.json();
//         })
//         .then((myJson) => {
//             console.log('myJson', myJson);
//             setCities(myJson)
//         });
// }