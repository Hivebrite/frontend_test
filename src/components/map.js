import React, { useState, useEffect } from 'react';
import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet'
import styled from 'styled-components'

const MapWrapper = styled.div`
    width: calc(80% - 20px);
    height: 100%;
    margin-left: 20px;
    border: 1px solid lightgrey;
    border-radius: 3px;
`
const Map = (props) => {
    const [coordinates, setCoordinates] = useState([44.582076, 103.461760]);
    const [details, setDetails] = useState('');

    const DrawMap = () => {
        return (
            <MapContainer
                center={coordinates}
                zoom={13}
                scrollWheelZoom={true}
            >
                <TileLayer
                    attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
                    url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                />
                <Marker position={coordinates}>
                    <Popup>{details}</Popup>
                </Marker>
            </MapContainer>
        )
    };

    useEffect(() => {
        setCoordinates([props.city.latitude, props.city.longitude]);
        setDetails(`
            state: ${props.city.state}
            population: ${props.city.population}
            growth 00-13: ${props.city.growth_from_2000_to_2013}
        `);
        DrawMap();
    }, [props.city]); //eslint-disable-line

    return (
        <>
            <MapWrapper>
                <DrawMap />
            </MapWrapper>
        </>
    )

};

export default Map;
