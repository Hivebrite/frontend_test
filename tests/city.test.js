import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components'

import { Button } from '../src/components/city';

describe('City component', () => {
    it('should render correctly', () => {
        const tree = renderer.create(
            <Button />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});