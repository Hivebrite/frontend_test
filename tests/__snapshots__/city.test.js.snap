// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City component should render correctly 1`] = `
.c0 {
  width: 100%;
  height: 60px !important;
  border: 1px solid lightgrey;
  border-radius: 5px;
  text-align: center;
  margin-bottom: 10px;
  cursor: pointer;
  font-family: 'Helvetica';
  font-weight: bold;
  font-size: 14px;
}

.c0:hover {
  background-color: lightblue;
  color: white;
}

<button
  className="c0"
/>
`;
